### PR TITLE
HTM-733, HTM-736, HTM-739 and HTM-740

### PR DESCRIPTION
**Pull request by ivanqin-obs**
Description:

- (The receiver should not process voice command through orbsessioncallback) removed in HTM-740
- Send data directly to orbsession
- Renaming the voice command function
- Update onVoiceSendKeyAction to dispatch key event
- Remove dispatchKeyPressEvent and notifyKeyUp
- Move voice service receiver to TIS (as well as mainactivity in mockapp)
- Create interface default method for consoleLog in session callback

### CODE REVIEW

File: android/orblibrary/src/main/java/org/orbtv/orblibrary/OrbSession.java [hunk 1]
```diff
@@ -25,6 +25,7 @@ class OrbSession implements IOrbSession {
     private Bridge mBridge;
     private BrowserView mBrowserView;
     private DsmccClient mDsmccClient;
+    private String MOCK_NEW_MEDIA_ID = "urn:broadcaster:programme:1249863457643";
```
Comment (joshua-h) [inline android/orblibrary/src/main/java/org/orbtv/orblibrary/OrbSession.java:28]: Why would we have a "MOCK" property in the main ORB library?

File: android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java [hunk 2]
```diff
@@ -901,51 +838,12 @@ default void onReceiveError(int code, String message,
...
-    /**
-     * @since 204
-     *
-     * Called to send a send a key press event, from a voice command, to the application 
-     *
-     * @param action The index number of the intent, either pressing a button or showing a log
-     */
-    default boolean onSendKeyPressAction(Integer action) {
-        throw new UnsupportedOperationException("Unsupported 204 API.");
+    default void consoleLog(String log) {
```
Comment (joshua-h) [inline android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java:846]: If this is only used for 204, can we throw an exception if it's not implemented, and implement it in the mock204app instead, to be consistent with other methods.

**Review by joshua-h — CHANGES_REQUESTED** [review]

File: android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java [hunk 3]
```diff
@@ -901,51 +838,23 @@ default void onReceiveError(int code, String message,
...
-     * @since 204
-     *
-     * Called to send a send a key press event, from a voice command, to the application 
-     *
-     * @param action The index number of the intent, either pressing a button or showing a log
+     * @param name The media name got from voice command
+     * @return The media id of media name.
      */
-    default boolean onSendKeyPressAction(Integer action) {
+    default String get_MEDIA_ID(String name) {
```
Comment (joshua-h) [inline android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java:857]: Please use a consistent style, `e.g., getMediaId(String name)`

File: android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java [hunk 4]
```diff
@@ -901,51 +838,23 @@ default void onReceiveError(int code, String message,
...
-     * Request for the description of the current media on applications
+     * Prints the log message using Logcat with a default log level of log.d.
+     * Overrides if a window console exists.
      *
-     * @return true if this event has been handled, and false if not
+     * @param log The log message to be printed
      */
-    default boolean onRequestMediaDescription() {
+    default void consoleLog(String log) {
+        Log.d("IOrbSessionCallback", "consoleLog: " + log);
```
Comment (joshua-h) [inline android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java:847]: This Log.d can be deleted. This class should be overriden by an implementation.

File: android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java [hunk 5]
```diff
@@ -16,11 +16,11 @@
 
 package org.orbtv.orblibrary;
 
+import android.util.Log;
```
Comment (joshua-h) [inline android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java:19]: Can be deleted

**Review by joshua-h — CHANGES_REQUESTED** [review]

File: android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java (hunk 5 above)
Comment (ivanqin-obs) [reply android/orblibrary/src/main/java/org/orbtv/orblibrary/IOrbSessionCallback.java]: Have done these changes

**Pull request merged**